Editor/Components/GuidedDataCollectorEditor.cs

```diff
@@ -94,6 +98,16 @@ public override void OnInspectorGUI()
 
         }
 
+        private void StepToTargetEnum(bool iterateToNext = true)
```

Comment:
Move this to the component itself and make it public.

Reply:
this is now in the component, and just acts as a setter. The logic for stepping through the defined target enums is still in the editor, and can easily be put elsewhere, depending on what's required